sui/src/config.rs

```diff
@@ -279,16 +290,43 @@ impl GenesisConfig {
         num_authorities: usize,
         num_accounts: usize,
         num_objects_per_account: usize,
+        key_pairs: Option<Vec<KeyPair>>,
     ) -> Result<Self, anyhow::Error> {
-        let mut authorities = Vec::new();
+        assert!(
+            num_authorities > 0,
+            "num_authorities should be larger than 0"
+        );
+        let mut authorities = Vec::with_capacity(num_authorities);
         for _ in 0..num_authorities {
             // Get default authority config from deserialization logic.
             let mut authority = AuthorityPrivateInfo::deserialize(Value::String(String::new()))?;
             authority.db_path = working_dir
                 .join(AUTHORITIES_DB_NAME)
-                .join(encode_bytes_hex(&authority.key_pair.public_key_bytes()));
+                .join(encode_bytes_hex(&authority.public_key));
             authorities.push(authority)
         }
+        let authority_key_pair;
+        if let Some(keypairs) = key_pairs {
+            // Use key pairs if given
+            assert_eq!(
+                keypairs.len(),
+                num_authorities,
+                "Number of key pairs does not maych num_authorities"
+            );
+            authority_key_pair = keypairs[0].copy();
```

Comment:
So from reading this, `GenesisConfig::custom_genesis` needs at most one of the key pairs it receives as argument: the first one. 

Please refactor so that it only receives this one single key pair as argument (and a suitable number of public keys besides, if needed). 

This will obviate the need for at least `key_pair_clone` in `validator_preparer.rs`.